docs/api-reference/next.config.js/rewrites.md

```diff
@@ -40,7 +40,7 @@ module.exports = {
 }
 ```
 
-Rewrites are applied to client-side routing, a `<Link href="/about">` will have the rewrite applied in the above example.
+Rewrites are not applied to client-side routing (`Link`, `router.push`), unless [Middleware](/docs/advanced-features/middleware) is present.
```

Comment:
`rewrites` are applied client-side regardless of middleware, `redirects` are not unless middleware is configured to match for that specific path

Reply:
Looks like #40549 is a bug then. Will change the wording here :+1: